### DISCUSSION TITLE
关于antd组件的默认导出方式

### DISCUSSION THREAD

**Discussion by KotoriK** [ant-design/ant-design#39151] [Q&A]
看了antd的组件文件的导出方式以后感觉非常的疑惑。明明可以使用默认导出和命名导出，现在的做法却是修改导出的组件，在上面加上一些东西。

这样会不会对Tree Shake十分不友好，比如DirectoryTree和Tree是完全不同的组件，但是因为DirectoryTree附加在Tree的身上，所以导入Tree的时候必定会被DirectoryTree也导入。

ConfigProvider这边还有一个ConfigProvider.config，整个组件库的代码里面没有找到用这个config的地方，但是也是因为上述的原因跟着导出，顺带扯着整个```@ctrl/tiny-color(18.7kB minified)```一起导出

这种做法还可能会带来循环导入的问题，例如刚刚提到的DirectoryTree，它本身会引入Tree组件，然后Tree组件内导出默认组件的时候又会把DirectoryTree附加到它身上一起导出。

所以为什么antd采取了如此特色的导出方式啊？？？

**Comment by yoyo837** [comment]
`DirectoryTree`和`Tree`做到同一个组件文件夹内的.

**Reply by KotoriK** [reply]
但是现在的导出方式下，就算我只想用Tree，DirectoryTree的代码也会被打包进来？

**Comment by yoyo837** [comment]
是的, 除非拆成两个, 其中一个依赖另一个. 放在一起个人理解应该是考虑`DirectoryTree`仅仅是一个衍生, 不算一个独立组件.
